test/integration/targets/postgresql/tasks/postgresql_ext.yml

```
@@ -0,0 +1,227 @@
+  become: yes
+  postgresql_query:
+    db: postgres
+    query: "DROP EXTENSION IF EXISTS postgis"
+  ignore_errors: yes
+
+##############
+# Start tests:
+
+# Create extension in check_mode, also check aliases for db and port params:
+- name: postgresql_ext - create extension postgis in check_mode
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_ext:
+    login_db: postgres
+    login_port: 5432
+    name: postgis
+  check_mode: yes
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that:
+    - result.changed == true
+    - result.queries == []
+
+# Check that extension doesn't exist after the previous step, rowcount must be 0
+- name: postgresql_ext - check that extension doesn't exist after the previous step
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_query:
+    db: postgres
+    query: "SELECT extname FROM pg_extension WHERE extname='postgis'"
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that: 
+    - result.rowcount == 0
+
+# Create extension postgis, also check aliases for db and port params
+- name: postgresql_ext - create extension postgis
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_ext:
+    login_db: postgres
+    login_port: 5432
+    name: postgis
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that:
+    - result.changed == true
+    - result.queries == ["CREATE EXTENSION \"postgis\""]
+
+# Check that extension exists after the previous step, rowcount must be 1
+- name: postgresql_ext - check that extension exists after the previous step
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_query:
+    db: postgres
+    query: "SELECT extname FROM pg_extension WHERE extname='postgis'"
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that: 
+    - result.rowcount == 1
+
+# Drop extension postgis:
+- name: postgresql_ext - drop extension postgis
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_ext:
+    db: postgres
+    name: postgis
+    state: absent
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that:
+    - result.changed == true
+    - result.queries == ["DROP EXTENSION \"postgis\""]
+
+# Check that extension doesn't exist after the previous step, rowcount must be 0
+- name: postgresql_ext - check that extension doesn't exist after the previous step
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_query:
+    db: postgres
+    query: "SELECT extname FROM pg_extension WHERE extname='postgis'"
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that: 
+    - result.rowcount == 0
+
+# Create extension postgis in particular schema
+- name: postgresql_ext - create extension postgis
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_ext:
+    db: postgres
+    name: postgis
+    schema: schema1
+  ignore_errors: yes
+  register: result
+
+- assert:
+    that:
+    - result.changed == true
+    - result.queries == ["CREATE EXTENSION \"postgis\" WITH SCHEMA \"schema1\""]
```

Possibly slightly more readable version:
```suggestion
    - result.queries == ['CREATE EXTENSION "postgis" WITH SCHEMA "schema1"']
```

---

fixed, thank you